index.js

```diff
@@ -106,10 +106,29 @@ function _walkSync(baseDir, options, _relativePath) {
     var entry = sortedEntries[i];
 
     if (entry.isDirectory()) {
-      if (options.directories !== false && (!globMatcher || globMatcher.match(entry.relativePath))) {
-        results.push(entry);
+
+      var realPath = fs.realpathSync(baseDir + '/' + entry.relativePath);
```

Comment:
It’s a bit unfortunate that in order to bust cycles we have to do an extra I/O operation per directory.

Reply:
ya :(